### PR TITLE
Automatic selected theme based on device

### LINKED ISSUE #4268: Add a premium setting to follow the system's theme where it is supported
## Abstract

Provide an option that when enabled will track the system's dark/light theme when supported by the OS, and will adjust the dark mode automatically based on that.

For more information check https://www.electronjs.org/docs/latest/tutorial/dark-mode

We need a new interop interface to track the status of the system theme.

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #4268

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: frontend/app/src/premium/premium-apis.ts [hunk 1]
```diff
@@ -89,8 +90,8 @@ export const userSettings = (): UserSettingsApi => {
     scrambleData: computed<boolean>(
       () => (store.state as any).session.scrambleData
     ),
-    darkModeEnabled: computed<boolean>(() => {
-      return store.getters['settings/darkModeEnabled'];
+    selectedTheme: computed<Theme>(() => {
+      return store.getters['settings/selectedTheme'];
```
Comment (kelsos) [inline frontend/app/src/premium/premium-apis.ts:94]: We need to find some time at some point to do a settings pinia/composition refactor.

**Review by kelsos — COMMENTED** [review]
I need to check the Linux part to see why it didn't adjust.

File: frontend/app/src/premium/premium-apis.ts (hunk 1 above)
Comment (lukicenturi) [reply frontend/app/src/premium/premium-apis.ts:94]: Strongly agree 😂

File: frontend/app/src/electron-main/ipc-setup.ts [hunk 2]
```diff
@@ -115,14 +115,13 @@ function setupVersionInfo() {
...
-      nativeTheme.themeSource = 'dark';
-    } else {
-      nativeTheme.themeSource = 'light';
-    }
-    event.sender.send(IPC_DARK_MODE, nativeTheme.shouldUseDarkColors);
+function setupSelectedTheme() {
+  ipcMain.on(IPC_THEME, async (event, selectedTheme: number) => {
+    const source = ['dark', 'system', 'light'][selectedTheme];
+    // @ts-ignore
+    nativeTheme.themeSource = source;
```
[the remark this answers is not in the document]
Comment (kelsos) [reply frontend/app/src/electron-main/ipc-setup.ts:122]: OK, so it seems to be a weird thing with Gnome 42, where you have to change two things for electron to work properly. 

The simple dark setting works with firefox, but I also have to set adwaita dark as the theme for chrome/electron to pick up the dark theme.

This is not our issue and if anyone asks we know why it happens.

**Review by kelsos — APPROVED** [review]
Looks good @lukicenturi after we merge it we need to update the tied commit on the components PR and merge that too.

We also need changelog entries I guess, both here and in the components

**Comment by lukicenturi** [comment]
> Looks good @lukicenturi after we merge it we need to update the tied commit on the components PR and merge that too.
> 
> We also need changelog entries I guess, both here and in the components

Added. By the way, how is the condition in Linux?

**Comment by kelsos** [comment]
> > Looks good @lukicenturi after we merge it we need to update the tied commit on the components PR and merge that too.
> > We also need changelog entries I guess, both here and in the components
> 
> Added. By the way, how is the condition in Linux?

Two separate things affect the theme on Gnome 42, I need to do the legacy dark theme for electron/chrome but it works:

https://github.com/rotki/rotki/pull/4460#discussion_r894370501

**Pull request merged**